### PR TITLE
Fix for CU-2xgnnvt and CU-2xpjf2x.

### PR DESCRIPTION
**Pull request by vivekvpandya**
Check asset_id before performing exchange().
Update tests accordingly.
Remove some unused imports.

## Issue
https://app.clickup.com/t/2xgnnvt
https://app.clickup.com/t/2xpjf2x

## Checklist

~- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_~
~- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_~

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2xgnnvt Users can swap between illegal asset pairs in Constant Product Pool](https://app.clickup.com/t/2xgnnvt)

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2xpjf2x Users can execute buy, sell orders on assetId's that is not listed in the pool](https://app.clickup.com/t/2xpjf2x)

**Review by vimukthi-git — COMMENTED** [review]
LGTM

File: frame/pablo/src/liquidity_bootstrapping.rs [hunk 1]
```diff
@@ -150,6 +150,10 @@ impl<T: Config> LiquidityBootstrapping<T> {
 		asset_id: T::AssetId,
 		amount: T::Balance,
 	) -> Result<T::Balance, DispatchError> {
+		ensure!(
+			asset_id == pool.pair.base || asset_id == pool.pair.quote,
```
[the remark this answers is not in the document]
Comment (vivekvpandya) [reply frame/pablo/src/liquidity_bootstrapping.rs:154]: Do you mean add such check method in each StableSwapPoolInfo,... ?

[the remark this answers is not in the document]
Comment (vimukthi-git) [reply frame/pablo/src/liquidity_bootstrapping.rs:154]: no no something like,
```suggestion
			pool.contains(asset_id),
```
above, where `contains` is a new method

[the remark this answers is not in the document]
Comment (hussein-aitlahcen) [reply frame/pablo/src/liquidity_bootstrapping.rs:154]: Please let's add this

[the remark this answers is not in the document]
Comment (vivekvpandya) [reply frame/pablo/src/liquidity_bootstrapping.rs]: I added `contains` in CurrencyPair. Please review.

**Review by hussein-aitlahcen — APPROVED** [review]

File: frame/composable-traits/src/defi.rs [hunk 2]
```diff
@@ -96,10 +96,14 @@ impl<AssetId: PartialEq> PartialEq for CurrencyPair<AssetId> {
 
 impl<AssetId: PartialEq> Eq for CurrencyPair<AssetId> {}
 
-impl<AssetId: Copy> CurrencyPair<AssetId> {
+impl<AssetId: Copy + PartialEq> CurrencyPair<AssetId> {
```
Comment (vimukthi-git) [inline frame/composable-traits/src/defi.rs:99]: :+1:

**Review by vimukthi-git — APPROVED** [review]
LGTM

**Pull request merged**